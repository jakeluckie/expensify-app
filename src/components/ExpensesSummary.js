import React from 'react'
import { connect } from 'react-redux'
import numeral from 'numeral'
import selectExpenses from '../selectors/expenses'
import selectExpensesTotal from '../selectors/expenses-total.js'

export const ExpensesSummary = ({ expenseCount, expenseTotal }) => {
    const expenseWord = expenseCount === 1 ? 'expense' : 'expenses'
    const formattedExpensesTotal = numeral(expenseTotal / 100).format('$0,0.00')

    return (
        <div>
            <h1>Viewing {expenseCount} {expenseWord} totalling {formattedExpensesTotal}</h1>
        </div>
    )
}

const mapStateToProps = (state) => {
    const filteredExpenses = selectExpenses(state.expenses, state.filters)
    return {
        expenseCount: filteredExpenses.length,
        expenseTotal: selectExpensesTotal(filteredExpenses)
    }
}

export default connect(mapStateToProps)(ExpensesSummary)