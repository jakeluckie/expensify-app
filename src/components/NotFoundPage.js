import React from 'react'
import { Link } from 'react-router-dom'

const NotFoundPage = () => (
    <div>
        404! Where'd he go? - <Link to="/">Go Home</Link>
    </div>
)

export default NotFoundPage