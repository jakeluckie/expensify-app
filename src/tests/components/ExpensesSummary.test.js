import React from 'react'
import { shallow } from 'enzyme'
import { ExpensesSummary } from '../../components/ExpensesSummary'
import expenses from '../fixtures/expenses'
import selectExpensesTotal from '../../selectors/expenses-total'

test('should display correct message for multiple expenses', () => {
    const wrapper = shallow(<ExpensesSummary 
        expenses={expenses}
        expenseCount={expenses.length}
        expenseTotal={selectExpensesTotal(expenses)}
    />)
    expect(wrapper).toMatchSnapshot()
})

test('should display correct message for a single expense', () => {
    const expense = [{
        id: '1',
        description: 'burn',
        note: '',
        amount: 2500,
        createdAt: 0
    }]
    const wrapper = shallow(<ExpensesSummary 
        expenses={expense}
        expenseCount={expense.length}
        expenseTotal={selectExpensesTotal(expense)}
    />)
    expect(wrapper).toMatchSnapshot()
})