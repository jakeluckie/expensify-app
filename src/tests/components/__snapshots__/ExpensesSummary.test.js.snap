// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should display correct message for a single expense 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totalling 
    $25.00
  </h1>
</div>
`;

exports[`should display correct message for multiple expenses 1`] = `
<div>
  <h1>
    Viewing 
    3
     
    expenses
     totalling 
    $90.00
  </h1>
</div>
`;
